src/include/OpenImageIO/imageio.h

```diff
@@ -416,10 +532,45 @@ class OIIO_API ImageSpec {
 
     /// Return the index of the named channel, or -1 if not found.
     int channelindex (string_view name) const;
+
+    /// Return pixel data window for this ImageSpec as a ROI.
+    ROI roi () const {
+        return ROI (x, x+width, y, y+height, z, z+depth, 0, nchannels);
+    }
+
+    /// Return full/display window for this ImageSpec as a ROI.
+    ROI roi_full () const {
+        return ROI (full_x, full_x+full_width, full_y, full_y+full_height,
+                    full_z, full_z+full_depth, 0, nchannels);
+    }
+
+    /// Set pixel data window for this ImageSpec to a ROI.
+    /// Does NOT change the channels of the spec, regardless of newroi.
```

Comment:
I'm sure you have a good rationale for ignoring the channel count, but it seems a bit asymmetric.

Perhaps it deserves a bit more explanation in the comment so the user understands why the chbegin/end doesn't get used?

Reply:
The short answer is that I just moved these from imagebuf.h/imagebuf.cpp to imageio.h, I didn't really think about their behavior.

It's a good question, though. The current use pattern is to use an ROI to set the x/y/z dimensions. I don't know how careful the callers are about the channel range or what it would break. I'll see what happens, and at worst, just make the docs clearer.